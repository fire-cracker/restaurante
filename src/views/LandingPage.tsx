import React from 'react'
import Container from 'react-bootstrap/Container'
import Jumbotron from 'react-bootstrap/Jumbotron'
import Col from 'react-bootstrap/Col'
import Row from 'react-bootstrap/Row'
import ResponsiveEmbed from 'react-bootstrap/ResponsiveEmbed'

const LandingPage: React.FC = () => {
  return (
    <Jumbotron fluid className='p-0'>
      <Container fluid>
        <Row>
          <ResponsiveEmbed aspectRatio='1by1'>
            <Col className='banner d-flex flex-column justify-content-center align-items-center'>
              <div className='text-0 text-darkkhaki'>
                <p>Restaurante Exquisite Cuisine</p>
              </div>
              <div className='text-1 text-white'>
                <p className='font-weight-normal'>Enjoy our Food Experience</p>
              </div>
              <div className='divider text-darkkhaki d-flex flex-row'>
                <p className='line my-2 mx-2'>&nbsp;</p>
                <p className='cutlery-icon'><i className="fa fa-cutlery"></i></p>
                <p className='line my-2 mx-2'>&nbsp;</p>
              </div>
              <div className='text-2 text-white'>
                <p>OPENING HOURS</p>
              </div>
              <div className='text-3 text-white d-flex flex-row'>
                <p className='days'>Monday - Friday:&nbsp;</p>
                <p className='time text-dimgray'>&nbsp;07:30AM - 23:00PM</p>
              </div>
              <div className='text-4 text-white d-flex flex-row'>
                <p className='days'>Weekends:&nbsp;</p>
                <p className='time text-dimgray'>&nbsp;07:30AM - 03:00PM</p>
              </div>
            </Col>
          </ResponsiveEmbed>
        </Row>
      </Container>
    </Jumbotron>
  )
}

export default LandingPage
