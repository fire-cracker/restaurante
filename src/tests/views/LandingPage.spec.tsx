import React, { RefObject } from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import LandingPage from '../../views/LandingPage'

const createMockStore = configureMockStore([thunk])
const store = createMockStore({})

interface Props {
  menuRef: RefObject<HTMLInputElement>
}

jest.mock('react-router', () => ({
  useParams: jest.fn().mockReturnValue({ id: '123' })
}))

describe('LandingPage', () => {
  const props: Props = {
    menuRef: React.createRef()
  }

  let wrapper: any
  beforeAll(() => {
    wrapper = render(
      <Provider store={store}>
        <LandingPage {...props} />
      </Provider>
    )
  })

  afterAll(() => {
    jest.clearAllMocks()
  })

  test('renders App component', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
