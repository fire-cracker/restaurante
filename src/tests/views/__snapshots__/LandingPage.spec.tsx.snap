// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders LandingPage component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-wrapper p-0 container-fluid"
      >
        <div
          class="p-0 jumbotron jumbotron-fluid"
        >
          <div
            class="container-fluid"
          >
            <div
              class="row"
            >
              <div
                class="embed-responsive embed-responsive-1by1"
              >
                <div
                  class="banner d-flex flex-column justify-content-center align-items-center embed-responsive-item col"
                >
                  <div
                    class="text-0 text-darkkhaki"
                  >
                    <p>
                      Restaurante Exquisite Cuisine
                    </p>
                  </div>
                  <div
                    class="text-1 text-white"
                  >
                    <p
                      class="font-weight-normal"
                    >
                      Enjoy our Food Experience
                    </p>
                  </div>
                  <div
                    class="divider text-darkkhaki d-flex flex-row"
                  >
                    <p
                      class="line my-2 mx-2"
                    >
                       
                    </p>
                    <p
                      class="icon"
                    >
                      <i
                        class="fa fa-cutlery"
                      />
                    </p>
                    <p
                      class="line my-2 mx-2"
                    >
                       
                    </p>
                  </div>
                  <div
                    class="text-2 text-white"
                  >
                    <p>
                      OPENING HOURS
                    </p>
                  </div>
                  <div
                    class="text-3 text-white d-flex flex-row"
                  >
                    <p
                      class="days"
                    >
                      Monday - Friday: 
                    </p>
                    <p
                      class="time text-dimgray"
                    >
                       07:30AM - 23:00PM
                    </p>
                  </div>
                  <div
                    class="text-4 text-white d-flex flex-row"
                  >
                    <p
                      class="days"
                    >
                      Weekends: 
                    </p>
                    <p
                      class="time text-dimgray"
                    >
                       07:30AM - 03:00PM
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="menu-jumbotron bg-ghostwhite px-0 jumbotron jumbotron-fluid"
        >
          <div
            class="menu-container m-auto container"
          >
            <div
              class="container-wrapper px-0 row"
            >
              <div
                class="container-body m-0 d-flex flex-column justify-content-center align-items-center container"
              >
                <div
                  class="menu-header mb-4 flex-column justify-content-center align-items-center row"
                >
                  <div
                    class="col"
                  >
                    <h4
                      class="text-darkkhaki"
                    >
                      FOOD AT FIRST SIGHT
                    </h4>
                  </div>
                  <div
                    class="col"
                  >
                    <h2>
                      Our Restaurant Menu
                    </h2>
                  </div>
                  <div
                    class="divider mt-1 mb-3 text-darkkhaki d-flex flex-row justify-content-center align-items-center col"
                  >
                    <p
                      class="line my-2 mx-1"
                    >
                       
                    </p>
                    <p
                      class="icon"
                    >
                      <img
                        alt="entwined"
                        class="fa fa-cutlery"
                        src="/images/entwined.svg"
                      />
                    </p>
                    <p
                      class="line my-2 mx-1"
                    >
                       
                    </p>
                  </div>
                </div>
                <div
                  class="navs-container mb-4 w-100 row"
                >
                  <div
                    class="w-100 justify-content-between nav"
                    role="tablist"
                  >
                    <div
                      class="nav-col p-0 col-lg-3 col-md-5"
                    >
                      <div
                        class="mr-4 mt-1 mb-0 p-0 nav-item"
                      >
                        <a
                          aria-selected="true"
                          class="py-4 px-3 bg-white nav-link active"
                          data-rb-event-key="breakfast"
                          href="#"
                          role="tab"
                        >
                          <img
                            alt="breafast icon"
                            src="/images/breakfast.svg"
                          />
                          <p>
                            Breakfast
                          </p>
                        </a>
                      </div>
                    </div>
                    <div
                      class="nav-col p-0 col-lg-3 col-md-5"
                    >
                      <div
                        class="mr-4 mt-1 mb-0 p-0 nav-item"
                      >
                        <a
                          aria-selected="false"
                          class="py-4 px-3 bg-white nav-link"
                          data-rb-event-key="lunch"
                          href="#"
                          role="tab"
                          tabindex="-1"
                        >
                          <img
                            alt="breafast icon"
                            src="/images/lunch.svg"
                          />
                          <p>
                            Lunch
                          </p>
                        </a>
                      </div>
                    </div>
                    <div
                      class="nav-col p-0 col-lg-3 col-md-5"
                    >
                      <div
                        class="mr-4 mt-1 mb-0 p-0 nav-item"
                      >
                        <a
                          aria-selected="false"
                          class="py-4 px-3 bg-white nav-link"
                          data-rb-event-key="dinner"
                          href="#"
                          role="tab"
                          tabindex="-1"
                        >
                          <img
                            alt="breafast icon"
                            src="/images/dinner.svg"
                          />
                          <p>
                            Dinner
                          </p>
                        </a>
                      </div>
                    </div>
                    <div
                      class="nav-col mr-0 p-0 col-lg-3 col-md-5"
                    >
                      <div
                        class="mr-0 mt-1 mb-0 p-0 nav-item"
                      >
                        <a
                          aria-selected="false"
                          class="py-4 px-3 bg-white nav-link"
                          data-rb-event-key="drink"
                          href="#"
                          role="tab"
                          tabindex="-1"
                        >
                          <img
                            alt="breafast icon"
                            src="/images/drink.svg"
                          />
                          <p>
                            Drinks
                          </p>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="menu-body mx-0 pt-5 row"
                >
                  <div
                    class="tab-content"
                  >
                    <div
                      aria-hidden="false"
                      class="fade tab-pane active show"
                      role="tabpanel"
                    >
                      <div
                        class="row"
                      />
                    </div>
                    <div
                      aria-hidden="true"
                      class="fade tab-pane"
                      role="tabpanel"
                    >
                      <div
                        class="row"
                      />
                    </div>
                    <div
                      aria-hidden="true"
                      class="fade tab-pane"
                      role="tabpanel"
                    >
                      <div
                        class="row"
                      />
                    </div>
                    <div
                      aria-hidden="true"
                      class="fade tab-pane"
                      role="tabpanel"
                    >
                      <div
                        class="row"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-wrapper p-0 container-fluid"
    >
      <div
        class="p-0 jumbotron jumbotron-fluid"
      >
        <div
          class="container-fluid"
        >
          <div
            class="row"
          >
            <div
              class="embed-responsive embed-responsive-1by1"
            >
              <div
                class="banner d-flex flex-column justify-content-center align-items-center embed-responsive-item col"
              >
                <div
                  class="text-0 text-darkkhaki"
                >
                  <p>
                    Restaurante Exquisite Cuisine
                  </p>
                </div>
                <div
                  class="text-1 text-white"
                >
                  <p
                    class="font-weight-normal"
                  >
                    Enjoy our Food Experience
                  </p>
                </div>
                <div
                  class="divider text-darkkhaki d-flex flex-row"
                >
                  <p
                    class="line my-2 mx-2"
                  >
                     
                  </p>
                  <p
                    class="icon"
                  >
                    <i
                      class="fa fa-cutlery"
                    />
                  </p>
                  <p
                    class="line my-2 mx-2"
                  >
                     
                  </p>
                </div>
                <div
                  class="text-2 text-white"
                >
                  <p>
                    OPENING HOURS
                  </p>
                </div>
                <div
                  class="text-3 text-white d-flex flex-row"
                >
                  <p
                    class="days"
                  >
                    Monday - Friday: 
                  </p>
                  <p
                    class="time text-dimgray"
                  >
                     07:30AM - 23:00PM
                  </p>
                </div>
                <div
                  class="text-4 text-white d-flex flex-row"
                >
                  <p
                    class="days"
                  >
                    Weekends: 
                  </p>
                  <p
                    class="time text-dimgray"
                  >
                     07:30AM - 03:00PM
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="menu-jumbotron bg-ghostwhite px-0 jumbotron jumbotron-fluid"
      >
        <div
          class="menu-container m-auto container"
        >
          <div
            class="container-wrapper px-0 row"
          >
            <div
              class="container-body m-0 d-flex flex-column justify-content-center align-items-center container"
            >
              <div
                class="menu-header mb-4 flex-column justify-content-center align-items-center row"
              >
                <div
                  class="col"
                >
                  <h4
                    class="text-darkkhaki"
                  >
                    FOOD AT FIRST SIGHT
                  </h4>
                </div>
                <div
                  class="col"
                >
                  <h2>
                    Our Restaurant Menu
                  </h2>
                </div>
                <div
                  class="divider mt-1 mb-3 text-darkkhaki d-flex flex-row justify-content-center align-items-center col"
                >
                  <p
                    class="line my-2 mx-1"
                  >
                     
                  </p>
                  <p
                    class="icon"
                  >
                    <img
                      alt="entwined"
                      class="fa fa-cutlery"
                      src="/images/entwined.svg"
                    />
                  </p>
                  <p
                    class="line my-2 mx-1"
                  >
                     
                  </p>
                </div>
              </div>
              <div
                class="navs-container mb-4 w-100 row"
              >
                <div
                  class="w-100 justify-content-between nav"
                  role="tablist"
                >
                  <div
                    class="nav-col p-0 col-lg-3 col-md-5"
                  >
                    <div
                      class="mr-4 mt-1 mb-0 p-0 nav-item"
                    >
                      <a
                        aria-selected="true"
                        class="py-4 px-3 bg-white nav-link active"
                        data-rb-event-key="breakfast"
                        href="#"
                        role="tab"
                      >
                        <img
                          alt="breafast icon"
                          src="/images/breakfast.svg"
                        />
                        <p>
                          Breakfast
                        </p>
                      </a>
                    </div>
                  </div>
                  <div
                    class="nav-col p-0 col-lg-3 col-md-5"
                  >
                    <div
                      class="mr-4 mt-1 mb-0 p-0 nav-item"
                    >
                      <a
                        aria-selected="false"
                        class="py-4 px-3 bg-white nav-link"
                        data-rb-event-key="lunch"
                        href="#"
                        role="tab"
                        tabindex="-1"
                      >
                        <img
                          alt="breafast icon"
                          src="/images/lunch.svg"
                        />
                        <p>
                          Lunch
                        </p>
                      </a>
                    </div>
                  </div>
                  <div
                    class="nav-col p-0 col-lg-3 col-md-5"
                  >
                    <div
                      class="mr-4 mt-1 mb-0 p-0 nav-item"
                    >
                      <a
                        aria-selected="false"
                        class="py-4 px-3 bg-white nav-link"
                        data-rb-event-key="dinner"
                        href="#"
                        role="tab"
                        tabindex="-1"
                      >
                        <img
                          alt="breafast icon"
                          src="/images/dinner.svg"
                        />
                        <p>
                          Dinner
                        </p>
                      </a>
                    </div>
                  </div>
                  <div
                    class="nav-col mr-0 p-0 col-lg-3 col-md-5"
                  >
                    <div
                      class="mr-0 mt-1 mb-0 p-0 nav-item"
                    >
                      <a
                        aria-selected="false"
                        class="py-4 px-3 bg-white nav-link"
                        data-rb-event-key="drink"
                        href="#"
                        role="tab"
                        tabindex="-1"
                      >
                        <img
                          alt="breafast icon"
                          src="/images/drink.svg"
                        />
                        <p>
                          Drinks
                        </p>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="menu-body mx-0 pt-5 row"
              >
                <div
                  class="tab-content"
                >
                  <div
                    aria-hidden="false"
                    class="fade tab-pane active show"
                    role="tabpanel"
                  >
                    <div
                      class="row"
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    class="fade tab-pane"
                    role="tabpanel"
                  >
                    <div
                      class="row"
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    class="fade tab-pane"
                    role="tabpanel"
                  >
                    <div
                      class="row"
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    class="fade tab-pane"
                    role="tabpanel"
                  >
                    <div
                      class="row"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
